### ISSUE TITLE
[Dev UI] Convert multi-line strings to JSON-escaped quotes when pasting into input box

### ISSUE THREAD

**Issue by mbleigh** [firebase/genkit#747]
It would be really handy for the input editor if, when you paste a multiline string, it automatically converted it to a JSON string. So i paste:

```
hello world
i am not "valid JSON" per se
```

and it turns it into

```
"hello world\ni am not \"valid JSON\" per se"
```

I asked an AI to built a proof of concept: 🙂 https://codepen.io/mbleigh/pen/GRbmvrG

**Comment by MichaelDoyle** [comment]
@hritan you worked on a similar issue, on large strings in the input editor. Did your changes capture this case as well?

**Comment by hritan** [comment]
> [@hritan](https://github.com/hritan) you worked on a similar issue, on large strings in the input editor. Did your changes capture this case as well?

Yes, but there is a catch. The input that we get from the editor in the component is already JSON stringified so manipulating that is a challenge, cz we have to first parse it and it throws error parsing an invalid JSON.

So, if someone enters:
`hello world
i am not valid JSON per se`
 this will work

But the following might not work:
`hello world
i am not "valid JSON" per se`

**Comment by mjchristy** [comment]
Do we want to do anything to support this use case or are we good to close this request?

**Comment by MichaelDoyle** [comment]
We should support this, yes. I think @hritan original implementation did it on the backend. But I like the idea (i.e. in Bleigh's demo) of intercepting the paste event instead and converting it right then and there.

It would:

- String escape any special characters
- Double quote it

We should start with inputs where they are initially "empty" and the schema is defined as a "string".

That takes care of the most common papercut. Once we get into editing a JSON it gets gnarly.

**Comment by spideyashith** [comment]
Hi @MichaelDoyle, I'd like to work on this based on your January scoping.

My understanding of the plan:
- Intercept the paste event in the input editor (not a backend transform)
- On paste, string-escape special characters and wrap in double quotes
- Scope this to the initial case only: inputs that are currently empty and whose schema is defined as type "string"
- Not touching the case of pasting into an already-populated or complex JSON structure, per your note that it "gets gnarly"

I'll start by locating the input editor component and the paste handling in the Dev UI codebase, then follow up here with a plan before opening a PR. Let me know if this is still the direction you want, since it's currently in Needs Triage.

**Comment by MichaelDoyle** [comment]
Hey @spideyashith unfortunately the Dev UI source is not public on GH at
this time, so we won’t be able to take contributions there.
